test/SlowTests/Bugs/Caching/UseCachingInLazyTests.cs

```diff
@@ -100,14 +100,16 @@ public async Task LazilyLoadAsync_WhenTrackChangesAndDoesntChange_ShouldNotCreat
 
             if (createVersion2)
             {
-                using (var session = store.OpenAsyncSession())
-                {
-                    await session.StoreAsync(new Doc {Version = "2"}, docId);
-                    await session.SaveChangesAsync();
-                }
-            }
+                var amre = new AsyncManualResetEvent();
+                var observable = store.Changes().ForDocument(docId);
```

Comment:
This is still racy. You have `EnsureSubscribedNow` method that you need to execute before doing any of actions below.